### ISSUE TITLE
Need more info to create TensorShape in each kernel

### ISSUE THREAD

**Issue by nhkrishna** [ARM-software/ComputeLibrary#762]
Hi @GeorgeARM ,

I have few questions regarding [these ](https://github.com/ARM-software/ComputeLibrary/issues/47#issuecomment-543220596)kind of scenarios mentioned in this issue.
You mentioned  "ACL would expect the shape of the input to be the same with the shape of the output for the ActivationLayer"  in above comments. 

- But i have observed that in non-activation layers(layers which does not use NEActivationLayer) such as Norm(Cross channel/Batch norm), softmax layers has the similar restriction and we are creating the Tensor shape for these specific layers as previous layer tensor shape and worked fine for me. Does it mean along with activation kernels norm, softmax kernels also has this restriction?

- Also my understanding is that FC and LSTM layers always expects output ARM Tensor shape as the 2D and Convolution, Pooling layers always expects output ARM Tensor shape as the 4D. Could you clarify on my understanding?

Could please point me the any documentation saying this restriction so that it would helpful for me to create bug free kernels with correct output tensor shape.

Thanks,
Hari

_Originally posted by @nhkrishna in https://github.com/ARM-software/ComputeLibrary/issues/47#issuecomment-543220596_

**Comment by nhkrishna** [comment]
Hi @morgolock ,

I have raised this issue to get more understanding creating TensorShape for a particular kernal with out any issues. I have referred [this](https://github.com/ARM-software/ComputeLibrary/issues/47#issuecomment-543220596) issue, where it will talk about output tensor shape of the ActivationLayers,  saying "ACL would expect the shape of the input to be the same with the shape of the output for the ActivationLayer".

But i have observed similar restriction in other kernels(mentioned in first comment) and few kernels always expect  fixed tensor shape(second point in first comment). I just wanted to find  this restrictions in any documentation or header  file but unfortunately i couldn't find.  Could please let us know where i can find documentation/test data sets to create TensorShape for a particular kernel without any issue.

Thanks,
Hari

**Comment by morgolock** [comment]
Hi @nhkrishna 

It depends on the particular function you are dealing with. In general the first place to look for this kind of information would be the header file of the function/kernel in ACL:
https://github.com/ARM-software/ComputeLibrary/blob/master/arm_compute/runtime/NEON/functions/NEActivationLayer.h
https://github.com/ARM-software/ComputeLibrary/blob/master/arm_compute/core/NEON/kernels/NEActivationLayerKernel.h

If the header does not contain the infromation you are looking for, the second place to look for this is in the kernel's validate() method:
https://github.com/ARM-software/ComputeLibrary/blob/master/src/core/NEON/kernels/NEActivationLayerKernel.cpp#L84

In this particular case you could see ` ARM_COMPUTE_RETURN_ERROR_ON_MISMATCHING_SHAPES(input, output);`

Hope this helps.

**Comment by GeorgeARM** [comment]
Hello @nhkrishna,

As @morgolock mentioned you can either check the kernels' validation code or the actual documentation where we might specify the expected size and data type of the outputs. We don't have strong restrictions thus ACL should comply with the operator specifications that other frameworks like TFLite and AnrdoidNN have.
Hope this helps.

Closing the issue. Reopen if needed
